### PR TITLE
Fix double counting GeoTileGrid aggregation

### LINKED ISSUE #74352: [CI] GeoTileTilerTests testGeoGridSetValuesBoundingBoxes_BoundedGeoShapeCellValues failing
**Build scan:**
https://gradle-enterprise.elastic.co/s/uprgvd2spxyug/tests/:x-pack:plugin:spatial:test/org.elasticsearch.xpack.spatial.search.aggregations.bucket.geogrid.GeoTileTilerTests/testGeoGridSetValuesBoundingBoxes_BoundedGeoShapeCellValues

**Reproduction line:**
`./gradlew ':x-pack:plugin:spatial:test' --tests "org.elasticsearch.xpack.spatial.search.aggregations.bucket.geogrid.GeoTileTilerTests.testGeoGridSetValuesBoundingBoxes_BoundedGeoShapeCellValues" -Dtests.seed=6A0F1C57DB02F8F8 -Dtests.locale=ru-RU -Dtests.timezone=Asia/Ho_Chi_Minh -Druntime.java=11 -Dtests.fips.enabled=true`

**Applicable branches:**
master, 7.x

**Reproduces locally?:**
Yes

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.spatial.search.aggregations.bucket.geogrid.GeoTileTilerTests&tests.test=testGeoGridSetValuesBoundingBoxes_BoundedGeoShapeCellValues

**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: <10>
     but: was <12>

  at __randomizedtesting.SeedInfo.seed([6A0F1C57DB02F8F8:667BAD9AC8E6C969]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.junit.Assert.assertThat(Assert.java:923)
  at org.elasticsearch.xpack.spatial.search.aggregations.bucket.geogrid.GeoGridTilerTestCase.testGeoGridSetValuesBoundingBoxes_BoundedGeoShapeCellValues(GeoGridTilerTestCase.java:123)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:566)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:938)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:974)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:988)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
  at java.lang.Thread.run(Thread.java:834)

```

### PR DESCRIPTION
**Pull request by iverase**
In the extreme case that a Geotile grid aggregation with a bounded box that crosses the dateline, and that bounding box edges are landing in the same tile, then it might happen that we are counting twice the same tile.

This PR makes sure this does not happen by adjusting which tiles we iterate for each side of the bounding box that crosses the dateline.

fixes https://github.com/elastic/elasticsearch/issues/74352

### CODE REVIEW

**Comment by iverase** [comment]
@elasticmachine run elasticsearch-ci/part-1

File: x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/search/aggregations/bucket/geogrid/GeoGridTilerTestCase.java [hunk 1]
```diff
@@ -124,6 +124,21 @@ public void testGeoGridSetValuesBoundingBoxes_BoundedGeoShapeCellValues() throws
         }
     }
 
+    // tests that bounding boxes that crosses the dateline and cover all longitude values are correctly wrapped
+    public void testGeoGridSetValuesBoundingBoxes_coversAlllongitudeValues() throws Exception {
+        int precision = 3;
+        Geometry geometry = new Rectangle(-92, 180, 0.99, -89);//GeometryTestUtils.randomRectangle();
```
Comment (imotov) [inline x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/search/aggregations/bucket/geogrid/GeoGridTilerTestCase.java:130]: Leftovers?

**Review by imotov — APPROVED** [review]
LGTM

File: x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/search/aggregations/bucket/geogrid/GeoGridTilerTestCase.java (hunk 1 above)
Comment (iverase) [reply x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/search/aggregations/bucket/geogrid/GeoGridTilerTestCase.java]: yes

**Pull request merged**